### ISSUE TITLE
Fixed Version in WPF: Retrieving html with JS after NavigationCompleted only works sporadically

### ISSUE THREAD

**Issue by Murlamue** [MicrosoftEdge/WebView2Feedback#629]
Hi there!

I'm using WebView2 in WPF to retrieve HTMLs of different websites, in fixed version mode. The WebView2-control-source is set with URIs out of a queue. After the NavigationCompleted event I inject JS:

`private async void WebView_NavigationCompleted(object sender, CoreWebView2NavigationCompletedEventArgs e)
        {
            var html = await _webView.ExecuteScriptAsync("document.documentElement.outerHTML;");
            ...
        }`
Then the next URI is read out of the queue and so on. 

**The problem:** Either all of the htmls are retrieved correctly or all of them incorrectly, meaning "null". It doesn't seem to be a problem of the js engine because if I inject another simple function like "Math.round(4.7);" after the ...outerHTML I get "5", no matter if I first got the html correctly or not. Timing also seems not to be the issue because the behavior keeps happening even after setting a timer to wait for a second after the actual NavigationCompleted event to get the html.

In short: I was not able to pinpoint when and why the html is retrieved incorrectly. Am I missing something, can someone help? Thanks!

**Version**
SDK: 1.0.674-prerelease
Runtime: Fixed Version
Framework: WPF
OS: Win10

**Comment by champnic** [comment]
Hm, that's very strange behavior. And to confirm, sometimes it all works fine, and sometimes it returns null for every url? In the cases when it returns null, are you able to see if it actually successfully navigated or not?

**Comment by Murlamue** [comment]
Tried it with the following code:

`private async void WebView_NavigationCompleted(object sender, CoreWebView2NavigationCompletedEventArgs e)
        {
            if (e.IsSuccess)
            {
                var html = await _webView.ExecuteScriptAsync("document.documentElement.outerHTML;");

                if (!string.IsNullOrEmpty(html) && !html.Equals("null"))
                {
                    //only doing something with the actual html 
                }
            }
        }`

Debugged it with actions instead of breakpoints to not mess with timing and watched out for NavigationStarted events, too. Apparentely every of my target urls fires up to 6 NavigationStarted and for most there is 1 NavigationCompleted. In some cases there are 2 NavigationCompleted events - in the first getting the html via js retrieves "null" and only the second one the actual html. I guess it in the end is a timing issue? I'm still fiddling though and it's no real explanation for the behavior where either all or no htmls where retrieved correctly.
